microsoft-edge/devtools-guide-chromium/memory-problems/memory-101.md

```diff
@@ -23,165 +23,146 @@ ms.date: 07/21/2023
 
-### Shallow size
 
-The _shallow size_ is the size of memory that is held by the object.
+<!-- ====================================================================== -->
+## JavaScript heap and renderer memory
 
-Typical JavaScript objects have some memory reserved for their description and for storing immediate values.  Usually, only arrays and strings can have a significant shallow size.  However, strings and external arrays often have their main storage in renderer memory, exposing only a small wrapper object on the JavaScript heap.
+The JavaScript heap is a memory region in the browser process where all the JavaScript objects live. The JavaScript heap is what the **Memory** tool displays when you take a heap snapshot. The JavaScript heap is also called the _V8 heap_ (after the V8 JavaScript engine that powers Microsoft Edge).
 
-_Renderer memory_ is all memory of the process where an inspected page is rendered:
+The JavaScript heap is part of the renderer memory. The renderer memory is the memory used by the browser process where the webpage is rendered. The renderer memory is comprised of the following:
 
-_renderer memory_ = _native memory_ + _JS heap memory of the page_ + _JS heap memory of all dedicated workers started by the page_
+* Native memory (such as the one used by the C++ objects that represent the DOM nodes).
+* JavaScript heap memory of the page.
+* JavaScript heap memory of all dedicated workers started by the page.
 
-Nevertheless, even a small object can hold a large amount of memory indirectly, by preventing other objects from being disposed of by the automatic garbage collection process.
 
-### Retained size
-
-The _retained size_ is the size of memory that is freed once the object is deleted along with the dependent objects that were made unreachable from garbage-collection roots (GC roots).
+<!-- ====================================================================== -->
+## Garbage-collection roots
 
-_Garbage-collection roots_ are made up of _handles_ that are created (as either local or global) when making a reference from native code to a JavaScript object outside of the V8 VM.  All such handles can be found within a heap snapshot under **GC roots** > **Handle scope** and **GC roots** > **Global handles**.  Describing the handles in this documentation without diving into details of the browser implementation may be confusing.  Both garbage-collection roots and the handles aren't something you need to worry about.
+_Garbage-collection roots_, or GC roots, are created by the browser when a reference from the browser's native code to a JavaScript object outside of the V8 virtual machine is made. These references are called _handles_. You can find these handles in the **Memory** tool under **GC roots** > **Handle scope** and **GC roots** > **Global handles**.<!-- Can't find this in the Memory tool. Where is it? -->
```

Comment:
@sethbrenith do you know if GC roots are still visible in the Memory tool, and if so, where?

Reply:
They're mostly hidden most of the time, and that's on purpose. By default, heap snapshots lie a little bit and pretend that any attached global objects (`Window`) are the roots. It's possible to see the term "GC roots" in the Containment view, but relatively little memory is attributed to that category.

However, the _idea_ of roots is somewhat important for understanding how a garbage collector works, which is the topic of a few paragraphs down from here.

Reply:
Thanks, removing the note about **Global handles** in DevTools, but keeping this intro about GC roots.